static/src/javascripts/projects/commercial/modules/article-aside-adverts.js

```diff
@@ -18,22 +20,23 @@ const calculateAllowedAdSlots = (availableSpace: number): string => {
     return '1,1|2,2';
 };
 
+const createSlotWrapper = (): Element => {
+    const adSlotWrapper = document.createElement('div');
+    adSlotWrapper.className = 'aside-slot-container js-aside-slot-container';
+    adSlotWrapper.setAttribute('aria-hidden', 'true');
+    return adSlotWrapper;
+};
+
 export const init = (start: () => void, stop: () => void): Promise<boolean> => {
     start();
 
     const $col: bonzo = $('.js-secondary-column');
-
-    // article aside ads are added server-side if the container doesn't exist then stop.
-    if (!$col.length || $col.css('display') === 'none') {
-        stop();
-        return Promise.resolve(false);
-    }
-
     const $mainCol: bonzo = $('.js-content-main-column');
     const $adSlot: bonzo = $('.js-ad-slot', $col);
     const $immersiveEls: bonzo = $('.element--immersive', $mainCol);
 
-    if (!$adSlot.length || !$mainCol.length) {
+    // article aside ads are added server-side UNLESS the page has a ShowcaseMainElement!
+    if (!$col.length || $col.css('display') === 'none') {
```

Comment:
Why was this moved below? the above `const` creations? I think you're using `$col` and doing a little bit where you don't need to. I'm assuming this is where some complexity has been remove, as you don't check for `adslot` or `maincol` anymore?

Reply:
Good point! Think I was just hacking around with the code, will switch it back. 👍